src/Common/Recognizer/TimeSurveyBotLuisEx.cs

```diff
@@ -69,18 +69,78 @@ public double WorkedDurationInMinutes(string culture = Culture.English)
             }
         }
 
-        public (DateTime start, DateTime end) WorkedPeriod(double minutes, string culture = Culture.English)
+        public (DateTime start, DateTime end) WorkedPeriod(IDateTimeProvider dateTimeProvider, double minutes,
+            string culture = Culture.English)
         {
             var workedPeriodInstances = Entities._instance.datetime;
             if (workedPeriodInstances.Length > 1)
             {
                 string? instance = workedPeriodInstances[1].Text;
-                var recognizedDateTime = DateTimeRecognizer.RecognizeDateTime(instance, culture).First();
+                var refTime = dateTimeProvider.DateTimeUtcNow();
+                var recognizedDateTime =
+                    DateTimeRecognizer.RecognizeDateTime(instance, culture, refTime: refTime).First();
                 var resolvedPeriod = ((List<Dictionary<string, string>>)recognizedDateTime.Resolution["values"])[0];
-                // TODO: use resolvedPeriod to pick a (start, end) period
+                return RecognizedWorkedPeriod(refTime, resolvedPeriod, minutes);
             }
-            var thisMorning = DateTime.Today.AddHours(9);
+
+            var thisMorning = dateTimeProvider.DateTimeUtcNow().Date.AddHours(9);
             return (thisMorning, thisMorning.AddMinutes(minutes));
         }
+
+        private static (DateTime start, DateTime end) RecognizedWorkedPeriod(DateTime refTime,
```

Comment:
What message is shown to the user in case this function fails to compute a worked period?
If it's the exception message I think we need to improve them

Reply:
It's the general exception message. Do you want to send a message to the user like "Invalid worked period provided, please use one of the following:
- insert x hours on project till 11 am
- insert x hours on project from 11 am
- insert x hours on project last Friday
- bla bla bla"

Reply:
no then it's ok. I just don't wanted it to be the specific exception message